--- a/linuxrc.syno.impl
+++ b/linuxrc.syno.impl
@@ -129,7 +129,6 @@ if [ "DVA1622" = "${HW_VERSION}" ]; then
        SYNOLoadHDMIAudio
 fi

-if [ "$SupportSAS" = "yes" ]; then
        echo "Insert SAS controller driver"
        #insert SAS6Gb driver
        if [ -e ${SAS2Driver} ]; then
@@ -145,7 +144,6 @@ if [ "$SupportSAS" = "yes" ]; then
        fi
        #wait 1 sec for drivers and disks to be ready
        sleep 1
-fi

 if [ "broadwell" = "$UniqueRD" -o "broadwellnk" = "$UniqueRD" -o "coffeelake" = "$UniqueRD" -o "skylaked" = "$UniqueRD" -o "v1000" = "$UniqueRD" -o "icelaked" = "$UniqueRD" ]; then
         echo "Insert Marvell 1475 SATA controller driver"
