--- a/linuxrc.syno
+++ b/linuxrc.syno
@@ -137,6 +139,7 @@
 # insert basic USB modules for detect f401/FDT
 echo "Insert basic USB modules..."
 SYNOLoadModules $USB_MODULES
+SYNOLoadModules "usb-storage"

 # insert Etron USB3.0 drivers

@@ -249,7 +252,7 @@
 fi
 if [ "$UniqueRD" = "kvmx64" -o \
 		"$UniqueRD" = "nextkvmx64" -o \
-		"$UniqueRD" = "kvmcloud" ]; then
+		"$UniqueRD" = "kvmcloud" ] || (grep -r -q -E "(QEMU|VirtualBox)" /sys/devices/virtual/dmi/id/); then
 	insmod /lib/modules/virtio.ko
 	insmod /lib/modules/virtio_ring.ko
 	insmod /lib/modules/virtio_pci.ko
