--- a/linuxrc.syno
+++ b/linuxrc.syno
@@ -153,6 +155,7 @@
 # insert basic USB modules for detect f401/FDT
 echo "Insert basic USB modules..."
 SYNOLoadModules $USB_MODULES
+SYNOLoadModules "usb-storage"
 
 # insert Etron USB3.0 drivers
 
@@ -276,7 +279,7 @@
 fi
 if [ "$UniqueRD" = "kvmx64" -o \
 		"$UniqueRD" = "nextkvmx64" -o \
-		"$UniqueRD" = "kvmcloud" ]; then
+		"$UniqueRD" = "kvmcloud" ] || (grep -r -q -E "(QEMU|VirtualBox)" /sys/devices/virtual/dmi/id/); then
 	insmod /lib/modules/virtio.ko
 	insmod /lib/modules/virtio_ring.ko
 	insmod /lib/modules/virtio_pci.ko
@@ -871,8 +874,11 @@
 # check if upgrade
 #
 if [ 0 -eq $FsckFailed ]; then
-    echo '------------upgrade'
-    /bin/sh /usr/syno/sbin/upgrade.sh
+  if echo '------------upgrade';then
+      /bin/sh /usr/syno/sbin/upgrade.sh
+  else
+      /bin/sh /usr/syno/sbin/upgrade.sh > /dev/null
+  fi
 	if [ $? -ne 0 ]; then
 		touch /.untar_upgrade_file_failed
 		Exit 6 "upgrade failed"
